src/components/ComponentPreview.tsx

```diff
@@ -40,18 +151,51 @@ const ComponentPreview: React.FC<ComponentPreviewProps> = ({component}) => {
         />
       </div>
       <div className="card-body" style={{maxHeight: '45vh', overflow: 'auto'}}>
-        <div className="component-preview" data-testid="componentPreview">
-          {previewMode === 'rich' ? (
-            <Preview component={component} />
-          ) : (
-            <JSONPreview data={component} />
-          )}
-        </div>
+        <Formik
+          enableReinitialize
+          initialValues={initialValues}
+          onSubmit={() => {
+            throw new Error("Can't submit preview form");
+          }}
+        >
```

Comment:
this makes a lot more sense to handle in a generic way, rather than burdening every component type with wrapping it themselves. It's also needed for proper multiple false/true toggle behaviour.

Reply:
Shouldn't this show some visual feedback for the user instead of just raising an error?